### PR TITLE
Feature/byte dance

### PR DESCRIPTION
**Pull request by Dogtiti**
## Summary by CodeRabbit

- **New Features**
  - Added support for ByteDance in the Settings component, allowing users to configure ByteDance endpoints and API keys.
  
- **Enhancements**
  - Introduced a new Doubao model provider for ByteDance, enabling integration with ByteDance services.
  - Updated chat components to display a more user-friendly model name for ByteDance models.
  - Improved localization with new ByteDance-related fields in both English and Chinese.

- **Configuration**
  - Added server-side configurations to support Bytedance API keys and URLs. 

- **Localization**
  - Added localization keys for ByteDance fields, including API key and endpoint, in both English and Chinese.

### CODE REVIEW

File: app/client/platforms/bytedance.ts [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+    let baseUrl = "";
+
+    if (accessStore.useCustomConfig) {
+      baseUrl = accessStore.bytedanceUrl;
+    }
+
+    if (baseUrl.length === 0) {
+      const isApp = !!getClientConfig()?.isApp;
+      baseUrl = isApp
+        ? DEFAULT_API_HOST + "/api/proxy/bytedance"
```
Comment (lloydzhou) [inline app/client/platforms/bytedance.ts:62]: 和百度的一样，这里也不需要走proxy，直接发请求就好了

File: app/client/platforms/bytedance.ts [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+    return [baseUrl, path].join("/");
+  }
+
+  extractMessage(res: any) {
+    return res.choices?.at(0)?.message?.content ?? "";
+  }
+
+  async chat(options: ChatOptions) {
+    const visionModel = isVisionModel(options.config.model);
```
Comment (lloydzhou) [inline app/client/platforms/bytedance.ts:83]: 也不用检查模型

File: app/client/platforms/bytedance.ts [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+            }
+          },
+          onmessage(msg) {
+            if (msg.data === "[DONE]" || finished) {
+              return finish();
+            }
+            const text = msg.data;
+            try {
+              const json = JSON.parse(text);
+              const choices = json.choices as Array<{
```
Comment (lloydzhou) [inline app/client/platforms/bytedance.ts:216]: 这里应该统一走  this.extractMessage函数

File: app/constant.ts [hunk 4]
```diff
@@ -104,6 +109,11 @@ export const Google = {
   ChatPath: (modelName: string) => `v1beta/models/${modelName}:generateContent`,
 };
 
+export const ByteDance = {
+  ExampleEndpoint: "https://ark.cn-beijing.volces.com/api/v3/chat/completions",
```
Comment (lloydzhou) [inline app/constant.ts:113]: 这个endpoint就不要v3/chat/completions

File: app/utils/model.ts [hunk 5]
```diff
@@ -50,9 +50,17 @@ export function collectModelTable(
         // 1. find model by name(), and set available value
         let count = 0;
         for (const fullName in modelTable) {
-          if (fullName.split("@").shift() == name) {
+          const [modelName, providerName] = fullName.split("@");
```
Comment (lloydzhou) [inline app/utils/model.ts:53]: 这里合并一下main分支看看，感觉写法上应该可以稍微优化一下

**Review by lloydzhou — COMMENTED** [review]
另外，这个pr缺少配置文档，需要在readme 以及readme_cn里面体现

**Review by lloydzhou — APPROVED** [review]

File: app/components/settings.tsx [hunk 6]
```diff
@@ -1249,6 +1250,51 @@ export function Settings() {
                       </ListItem>
                     </>
                   )}
+
+                  {accessStore.provider === ServiceProvider.ByteDance && (
+                    <>
+                      <ListItem
+                        title={Locale.Settings.Access.ByteDance.Endpoint.Title}
+                        subTitle={
+                          Locale.Settings.Access.Anthropic.Endpoint.SubTitle +
```
Comment (Dogtiti) [inline app/components/settings.tsx:1259]: ByteDance

**Pull request merged**